azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -289,17 +301,17 @@ def _execute(
 
         request_payload = {"db": kusto_database, "csl": kusto_query}
 
-        access_token = self._aad_helper.acquire_token()
         request_headers = {
-            "Authorization": access_token,
             "Accept": "application/json",
             "Accept-Encoding": "gzip,deflate",
             "Content-Type": "application/json; charset=utf-8",
-            "Fed": "True",
             "x-ms-client-version": "Kusto.Python.Client:" + VERSION,
             "x-ms-client-request-id": "KPC.execute;" + str(uuid.uuid4()),
         }
 
+        if self._aad_helper:

```

Comment:
this is awkward and hard to read the intent from. why wouldn't I add auth if I have no aad_helper? the actual reasin I don't use auth, is because no `auth_provider` or `auth_method` was set, which is what I would expect to see here.

Reply:
I agree. Can we control this behavior by auth_mehtod being not set or equal to "Anonymous"?